cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -87,6 +87,7 @@
   private static final String CPU_MULTIPLIER = "master.environment.k8s.container.cpu.multiplier";
   private static final String MEMORY_MULTIPLIER = "master.environment.k8s.container.memory.multiplier";
   private static final String DEFAULT_MULTIPLIER = "1.0";
+  private static final double MEMORY_MAX_HEAP_SIZE_FRACTION = 0.8;
```

Comment:
We have two keys in the cdap-site.xml to control non-heap memory, `twill.java.heap.memory.ratio` and `twill.java.reserved.memory.mb`. You can get those through the `MasterEnvironmentContext.getConfigurations()` method. You can refer to the `org.apache.twill.internal.utils.Resources.computeMaxHeapSize` method on how it computes the xmx.

Reply:
Done. Added twill as dependency and uses the util function directly.